bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiBridgeHandler.java

```diff
@@ -61,6 +69,11 @@ public ApiHandler getApi() {
     @Override
     public void dispose() {
         logger.debug("Disposing Freebox OS API handler for thing {}", getThing().getUID());
+        Future<?> job = openConnectionJob;
+        if (job != null && !job.isCancelled()) {
+            job.cancel(true);
+            openConnectionJob = null;
+        }
         apiHandler.closeSession();
```

Comment:
Ideally, it should be run inside a separate thread so that the dispose is very fast. But I finally did not do it because I have to be sure that "managers" will be cleared before initialize is called again.